doc/stack_vs_cabal_config.md

```diff
@@ -0,0 +1,135 @@
+<div class="hidden-warning"><a href="https://docs.haskellstack.org/"><img src="https://rawgit.com/commercialhaskell/stack/master/doc/img/hidden-warning.svg"></a></div>
+
+# stack vs cabal config files
+
+Due to their apparent overlap, the purpose of the following three files can be
+unclear:
+
+* `stack.yaml`
+* A cabal file, e.g. `my-package.cabal`
+* `package.yaml`
+
+The last two are easy to explain: `package.yaml` is a file format supported by
+[hpack](https://github.com/sol/hpack#readme), which adds some niceties on top
+of cabal like YAML syntax support and automatic generation of `exposed-modules`
```

Comment:
More correctly `deb-control` like file: https://www.debian.org/doc/manuals/maint-guide/dreq.en.html

Reply:
I don't understand this comment, sorry.

Reply:
Sorry, I missparsed the sentence:

> adds some niceties on top of (cabal like YAML syntax)....

The sentence is too long.

`stack` converts (using [hpack](https://github.com/sol/hpack#readme)) `package.yaml` package descriptions into equivalent `my-package.cabal` files. Hpack is designed to remove redundancies from your package descriptions, for example it uses YAML syntax and supports and automatic generateion of `exposed-modules`